src/index.test.js

```diff
@@ -125,6 +125,34 @@ describe('cmp.willShowPrivacyMessage', () => {
 	});
 });
 
+describe('cmp.willShowPrivacyMessageSync', () => {
+	it('throws if CMP is not initialised', () => {
+		expect(() => cmp.willShowPrivacyMessageSync()).toThrow();
+	});
+
+	it('does not throw if CMP is initialised', () => {
+		cmp.init({ country: 'GB' });
+
+		cmp.willShowPrivacyMessage().then(() => {
```

Comment:
It’s funny we’re still using the resolved state of `willShowPrivacyMessage` as a way to check the CMP has initialised.

Would it make sense for `init` to return a Promise?

Reply:
I think it's fine. Because everyone who cares is using onConsentChange. Init already returns a promise.
But if we wanted to I think the change would be this:

```
void frameworkCMP.willShowPrivacyMessage().then((willShowValue) => {
		_willShowPrivacyMessage = willShowValue;
		initComplete = true;
	}).then(resolveInitialised);
```

Reply:
`init` currently returns `void`, but I agree as this is internal use there's no need to really worry about it.